### PR TITLE
Transform modules

### PR DESCRIPTION
**Pull request by datashaman**
Adds the ability to use a Javascript module for the transform. When specifying the `transform` option, prefix the value with `@` (a curl convention) to load the top-level function which is called with the document and the parsed arguments to the module.

Uses a pseudo-URL format to specify arguments to the module as follows:

    elasticdump --transform='@./transforms/my-transform?param1=value&param2=another-value'

With a module at `./transforms/my-transform.js` with the following:

    module.exports = function (doc, options) {
        // do something to doc
    };

will load module `./transforms/my-transform.js', and execute the function with `doc` and `options` = `{"param1": "value", "param2": "another-value"}`.

Also supplied is an example transform for anonymizing data on-the-fly. It works well for our needs, and may suit yours too.

Regular scripts passed as strings to `elasticdump` are still parsed and used the same as before. This changes nothing about existing behaviour, only adds to it. Notably, this bypasses the security of the sandboxed vm environment used by the string-based transform, giving you more flexibility.

### CODE REVIEW

**Comment by evantahler** [comment]
This is great, but please remove the yarn lockfile.  we can discuss that as a separate issue.

**Comment by datashaman** [comment]
sure thing.

**Comment by evantahler** [comment]
👍

File: elasticdump.js [hunk 1]
```diff
@@ -81,8 +96,15 @@ var elasticdump = function (input, output, options) {
...
-      var modificationScriptText = '(function(doc) { ' + transform + ' })'
-      return new vm.Script(modificationScriptText).runInThisContext()
+      if (transform[0] === '@') {
+        return function (doc) {
+          var parsed = url.parse(transform.slice(1))
+          return require(parsed.pathname)(doc, getParams(parsed.query))
+        }
+      } else {
+        var modificationScriptText = '(function(doc) { ' + transform + ' })'
+        return new vm.Script(modificationScriptText).runInNewContext()
```
Comment (datashaman) [inline elasticdump.js]: Sorry, just realized that this reverted back to `runInNewContext`. I'll submit a PR now.

**Pull request merged**